ESP8266/ESP8266.cpp

```diff
@@ -215,7 +220,7 @@ void ESP8266::_packet_handler()
     uint32_t amount;
 
     // parse out the packet
-    if (!_parser.recv(",%d,%d:", &id, &amount)) {
+    if (!_parser.recv(",%d,%" SCNu32 ":" , &id, &amount)) {
```

Comment:
Not really. ld means long signed int. Proper way of doing it is this. I don't see any other proper alternatives.